saleor/graphql/product/dataloaders/attributes.py

```diff
@@ -69,24 +61,55 @@ def map_attributes(attributes):
         )
 
 
-class ProductAttributesByProductTypeIdLoader(
+class ProductAttributesAllByProductTypeIdLoader(
     BaseProductAttributesByProductTypeIdLoader
 ):
-    """Loads product attributes by product type ID."""
+    context_key = "product_attributes_all_by_producttype"
+    model_name = AttributeProduct
+
+    def get_queryset(self):
+        return self.model_name.objects.using(self.database_connection_name).all()
 
-    context_key = "product_attributes_by_producttype"
+
+class ProductAttributesVisibleInStorefrontByProductTypeIdLoader(
+    BaseProductAttributesByProductTypeIdLoader
+):
+    context_key = "product_attributes_visible_in_storefront_by_producttype"
     model_name = AttributeProduct
 
+    def get_queryset(self):
+        return self.model_name.objects.using(self.database_connection_name).filter(
+            attribute__visible_in_storefront=True
```

Comment:
Same here:
`Exists` and `OuterRef` should be used here, as `attribute__visible_in_storefront` is generating SQL with `inner join`